include/pybind11/cast.h

```diff
@@ -10,947 +10,11 @@
 
 #pragma once
 
-#include "pytypes.h"
-#include "detail/typeid.h"
-#include "detail/descr.h"
-#include "detail/internals.h"
-#include <array>
-#include <limits>
-#include <tuple>
-#include <type_traits>
-
-#if defined(PYBIND11_CPP17)
-#  if defined(__has_include)
-#    if __has_include(<string_view>)
-#      define PYBIND11_HAS_STRING_VIEW
-#    endif
-#  elif defined(_MSC_VER)
-#    define PYBIND11_HAS_STRING_VIEW
-#  endif
-#endif
-#ifdef PYBIND11_HAS_STRING_VIEW
-#include <string_view>
-#endif
-
-#if defined(__cpp_lib_char8_t) && __cpp_lib_char8_t >= 201811L
-#  define PYBIND11_HAS_U8STRING
-#endif
+#include "detail/type_caster_base.h"
```

Comment:
Not a full review, but a first drive-by comment: this is not really good include-what-you-use practice (referring to the principle, not the tool!).

Any chance of a quick scan on what part of the original includes is still used in the leftover file? This doesn't need to be exactly perfect (the rest of pybind11 isn't either), but depending on the includes of an included file is not amazing (as far as I know).

Reply:
See updated PR description (I actually used the `iwyu` tool).